src/main/java/org/openstreetmap/atlas/checks/distributed/IntegrityCheckSparkJob.java

```diff
@@ -196,34 +196,38 @@ public void start(final CommandMap commandMap)
         final Map<String, String> sparkContext = configurationMap();
         final CheckResourceLoader checkLoader = new CheckResourceLoader(checksConfiguration);
 
-        // Load checks
-        final Set<BaseCheck> checksToExecute = checkLoader.loadChecks();
-        if (!isValidInput(countries, checksToExecute))
+        // check configuration and country list
+        final Set<BaseCheck> preOverriddenChecks = checkLoader.loadChecks();
+        if (!isValidInput(countries, preOverriddenChecks))
         {
             logger.error("No countries supplied or checks enabled, exiting!");
             return;
         }
 
+        // Load checks
+        final Map<String, Set<BaseCheck>> checksToExecutePerCountry = checkLoader
+                .loadChecksForCountries(countries);
```

Comment:
Are collecting configuration for all the groups necessary here? I'd suggest updating `loadChecksForCountries` method to `loadChecksForGroup` to load checks with overridden group key. We should not hold onto such data structure here. Create as needed and pass in.

Reply:
Oh I see just load that during tuple creation.